modules/frontend/retry.go

```diff
@@ -37,6 +37,7 @@ func (r retryWare) Do(req *http.Request) (*http.Response, error) {
 	ctx := req.Context()
 	span, ctx := opentracing.StartSpanFromContext(ctx, "frontend.Retry")
 	defer span.Finish()
+	span.SetTag("span.kind", "client")
```

Comment:
I think it might be better to use the `opentracing-go/ext` which has the right set of tags to add 
Ex: https://github.com/weaveworks/common/blob/284892f3c53779d1a5956460b7fc8843b02ae7a9/instrument/instrument.go#L157

Reply:
Doesn't this already work with the cortex/loki? any idea why those are working? Don't we need a matching span.kind = server? or does that already exist?

Reply:
> Doesn't this already work with the cortex/loki? any idea why those are working?
Yes, it works for them. Cortex is using this [instrumentation](https://github.com/cortexproject/cortex/blob/master/pkg/querier/queryrange/instrumentation.go#L26-L30) for its middlewares, which instruments requests that go through them. I can explore that too.

> Don't we need a matching span.kind = server? or does that already exist?
Yeah, the globaltracer that Tempo uses correctly observes the incoming request and tags it accordingly.